### PR TITLE
Change to ignore `example?`, `hook?`, `let?`, `include?` and `subject?` matchers with `RSpec` receivers

### PR DESCRIPTION
**Pull request by ydah**
Follow up: https://github.com/rubocop/rubocop-rspec/pull/1576#issuecomment-1421499150

______________________________________________________________________

Before submitting the PR make sure the following are checked:

- [x] Feature branch is up-to-date with `master` (if not - rebase it).
- [x] Squashed related commits together.
- [-] Added tests.
- [-] Updated documentation.
- [-] Added an entry to the `CHANGELOG.md` if the new code introduces user-observable changes.
- [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

### CODE REVIEW

**Comment by pirj** [comment]
I think the same applies to `hook?`/`example?`/`subject?`/`include?`. Would you like to take care of all that in one go?

**Comment by ydah** [comment]
> I think the same applies to `hook?`/`example?`/`subject?`/`include?`. Would you like to take care of all that in one go?

Indeed it is. I updated this PR. Thank you.

File: lib/rubocop/rspec/language.rb [hunk 1]
```diff
@@ -44,30 +44,34 @@ class << self
...
+      def_node_matcher :example?, '(block (send nil? #Examples.all ...) ...)'
 
       # @!method hook?(node)
-      def_node_matcher :hook?,
-                       block_or_numblock_pattern('#Hooks.all')
+      def_node_matcher :hook?, <<-PATTERN
+      {
+        (numblock (send nil? #Hooks.all ...) ...)
+        (block (send nil? #Hooks.all ...) ...)
+      }
```
Comment (pirj) [inline lib/rubocop/rspec/language.rb:54]:
```suggestion
        {
          (numblock (send nil? #Hooks.all ...) ...)
          (block (send nil? #Hooks.all ...) ...)
        }
```

**Review by pirj — APPROVED** [review]
Thank you! Personally, those node patterns look less cryptic to me now 🤩

**Pull request closed without merging**